### PR TITLE
Updated highlights with Neovim V0.9 LSP highlighting.

### PR DESCRIPTION
**Pull request by Bats6789**
[Neovim V0.9.0](https://github.com/neovim/neovim/releases/tag/v0.9.0) added support for semantic highlighting using LSP's. This commit adds the additional highlight groups to support that feature. 

The highlight groups are only added if the user has V0.9 or higher. Previous version should exclude the additional groups, which may be unnecessary.

This commit also adds a table concatenation function in the utils folder. This function is used for adding the additional highlight groups.

### CODE REVIEW

**Comment by Bats6789** [comment]
> Maybe a stupid question, but I am curious that's why I am asking.
> 
> Is it possible to check the changes in highlighting after applying this change? Because I copied some code from [here](https://gist.github.com/swarn/fb37d9eefe1bc616c2a7e476c0bc0316#what-is-semantic-highlighting) and even after applying the changes to tokyodark colorscheme files I couldn't see any visible changes.

In the case of semantic highlighting, you must have an LSP that is capable of reporting semantic highlighting.

> Or is this some change that only will be noticeable when the a plugin or language specifically makes use of this semantic highlighting feature?

You should be able to see changes occur by using the hi command. Use the :Inspect command over some text you expect to receive the semantic highlighting and see if it has @lsp highlighting group.

**Comment by tiagovla** [comment]
Hey, just include the highlight groups directly under [here](https://github.com/tiagovla/tokyodark.nvim/blob/fe5847e379b3d1c52896fa69ff8faf20cde483cc/lua/tokyodark/highlights.lua#L323). Older versions will probably ignore it. If not, there's a version check function in utils, and table concatenation can be done using `vim.tbl_extend`.

**Comment by Bats6789** [comment]
> Hey, just include the highlight groups directly under [here](https://github.com/tiagovla/tokyodark.nvim/blob/fe5847e379b3d1c52896fa69ff8faf20cde483cc/lua/tokyodark/highlights.lua#L323). Older versions will probably ignore it. If not, there's a version check function in utils, and table concatenation can be done using `vim.tbl_extend`.

I moved the highlights to the suggested section.

File: lua/tokyodark/utils.lua [hunk 1]
```diff
@@ -44,4 +44,11 @@ function U.check_min_version(major, minor, patch)
     return vim.version.gt({ major, minor, patch }, vim.version())
 end
 
+function U.TableConcat(t1, t2)
```
Comment (tiagovla) [inline lua/tokyodark/utils.lua:47]: Just remove this and I will merge.

**Review by tiagovla — CHANGES_REQUESTED** [review]

**Review by tiagovla — APPROVED** [review]

**Pull request merged**